tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -0,0 +1,375 @@
+ * - NB11 Absolute ChargePointMaxProfile Profile with connector id not 0
+ * - NB12 Absolute TxProfile connector_id == 0
+ */
+class ChargepointTestFixtureV201 : public testing::Test {
+protected:
+    void SetUp() override {
+    }
+
+    ChargingSchedule createChargeSchedule() {
+        return ChargingSchedule{{}};
+    }
+
+    ChargingSchedule createChargeSchedule(ChargingRateUnitEnum chargingRateUnit) {
+        int32_t id;
+        std::vector<ChargingSchedulePeriod> chargingSchedulePeriod;
+        std::optional<CustomData> customData;
+        std::optional<ocpp::DateTime> startSchedule;
+        std::optional<int32_t> duration;
+        std::optional<float> minChargingRate;
+        std::optional<SalesTariff> salesTariff;
+
+        return ChargingSchedule{
+            id,          chargingRateUnit, chargingSchedulePeriod, customData, startSchedule, duration, minChargingRate,
+            salesTariff,
+        };
+    }
+
+    ChargingProfile createTxProfile(ChargingSchedule chargingSchedule, std::string transactionId, int stackLevel = 1) {
+        auto chargingProfileId = 1;
+        auto chargingProfilePurpose = ChargingProfilePurposeEnum::TxProfile;
+        auto chargingProfileKind = ChargingProfileKindEnum::Absolute;
+        auto recurrencyKind = RecurrencyKindEnum::Daily;
+        std::vector<ChargingSchedule> chargingSchedules = {chargingSchedule};
+        return ChargingProfile{.id = chargingProfileId,
+                               .stackLevel = stackLevel,
+                               .chargingProfilePurpose = chargingProfilePurpose,
+                               .chargingProfileKind = chargingProfileKind,
+                               .chargingSchedule = chargingSchedules,
+                               .customData = {},
+                               .recurrencyKind = recurrencyKind,
+                               .validFrom = {},
+                               .validTo = {},
+                               .transactionId = transactionId};
+    }
+
+    ChargingProfile createTxProfileWithMissingTransactionId(ChargingSchedule chargingSchedule) {
+        auto chargingProfileId = 1;
+        auto stackLevel = 1;
+        auto chargingProfilePurpose = ChargingProfilePurposeEnum::TxProfile;
+        auto chargingProfileKind = ChargingProfileKindEnum::Absolute;
+        auto recurrencyKind = RecurrencyKindEnum::Daily;
+        std::vector<ChargingSchedule> chargingSchedules = {chargingSchedule};
+        return ChargingProfile{
+            chargingProfileId,
+            stackLevel,
+            chargingProfilePurpose,
+            chargingProfileKind,
+            chargingSchedules,
+            {}, // transactionId
+            recurrencyKind,
+            {}, // validFrom
+            {}  // validTo
+        };
+    }
+
+    /**
+     * TxDefaultProfile, stack #1: time-of-day limitation to 2 kW, recurring every day from 17:00h to 20:00h.
```

Comment:
Are the comments in this file accurate? They appear to refer to `TxDefaultProfile`s instead of `TxProfile`.